crates/uv-configuration/src/threading.rs

```diff
@@ -0,0 +1,51 @@
+//! Configure rayon and determine thread stack sizes.
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::LazyLock;
+use uv_static::EnvVars;
+
+/// The default minimum stack size for uv threads.
+pub const UV_MIN_STACK_DEFAULT: usize = 4 * 1024 * 1024;
+
+/// Running out of stack has been an issue for us. We box types and futures in various places
+/// to mitigate this.
+///
+/// Main thread stack-size has a BIG variety here across platforms and it's harder to control
+/// (which is why Rust doesn't by default). Notably  on macOS and Linux you will typically get 8MB
+/// main thread, while on Windows you will typically get 1MB, which is *tiny*:
+/// <https://learn.microsoft.com/en-us/cpp/build/reference/stack-stack-allocations?view=msvc-170>
+///
+/// To normalize this we just spawn a new thread called main2 with a size we can set
+/// ourselves. 2MB is typically too small (especially for our debug builds), while 4MB
+/// seems fine. Also we still try to respect `RUST_MIN_STACK` if it's set, in case useful,
+/// but don't let it ask for a smaller stack to avoid messy misconfiguration since we
+/// know we use quite a bit of main stack space.
+///
+/// Non-main threads should all have 2MB, as Rust forces platform consistency there,
+/// but even then stack overflows can occur in release mode
+/// (<https://github.com/astral-sh/uv/issues/12769>), so also configure a 4MB stack for tokio and
+/// rayon threads.
+pub fn min_stack_size() -> usize {
+    std::env::var(EnvVars::RUST_MIN_STACK)
+        .ok()
+        .and_then(|var| var.parse::<usize>().ok())
+        .unwrap_or(UV_MIN_STACK_DEFAULT)
```

Comment:
the `unwrap_or(0).max(UV_MIN_STACK_DEFAULT)` was imo important for ensuring no errant/unrelated uses of RUST_MIN_STACK broke uv.

Reply:
Doesn't `unwrap_or(0).max(UV_MIN_STACK_DEFAULT)` mean that we never allow to set a smaller stack size, even with `RUST_MIN_STACK`? My idea was that we allow `RUST_MIN_STACK` in both directions, so that we can both test the effect of lower values (is the bound too high?) and allow bumping it higher (for testing in cases https://github.com/astral-sh/uv/issues/12769 like where we need to bump it higher).

Reply:
I put a 1MB limit in place and switched to `UV_STACK_SIZE`